### PR TITLE
Clear cache

### PR DESCRIPTION
**Pull request by rbracco**
Add functionality to view cache memory usage, and manually clear cache.

### CODE REVIEW

File: audio/data.py [hunk 1]
```diff
@@ -68,10 +68,40 @@ class AudioConfig:
...
+        num_removed = 0
+        if not os.path.exists(self.cache_dir/"cache_contents.txt"):
+            print("Cache contents not found, try calling again after creating your AudioList")
+            
+        with open(self.cache_dir/"cache_contents.txt", 'r') as f:
+            pb = progress_bar(f.read().split('\n')[:-1])
+            for line in pb:
+                if not os.path.exists(line): continue
+                else: 
+                    os.remove(line)
```
Comment (mogwai) [inline audio/data.py]: Need some  try catch blocks here?

File: audio/data.py [hunk 2]
```diff
@@ -68,10 +68,40 @@ class AudioConfig:
...
+                    if(os.path.exists(parent) and len(parent.ls()) == 0): os.rmdir(str(parent))
+        os.remove(self.cache_dir/"cache_contents.txt")
+        print(f"{num_removed} files removed")
+     
+    def cache_size(self):
+        '''Check cache size, returns a tuple of int in bytes, and string representing MB'''
+        cache_size = 0
+        if not os.path.exists(self.cache_dir):
+            print("Cache not found, try calling again after creating your AudioList")
+            return (None, None)
```
Comment (mogwai) [inline audio/data.py:111]: Perhaps here too?

File: audio/data.py [hunk 3]
```diff
@@ -136,8 +168,15 @@ def segment_items(item, config, path):
             if((i+1)*segsize <= siglen): sigs.append(sig[i*segsize:(i+1)*segsize])
             else: sigs.append(torch.cat([sig[i*segsize:], torch.zeros(segsize-len(sig[i*segsize:]))]))
         files = make_cache(sigs, sr, config, "s", item_path, [config.segment_size])
+        _record_cache_contents(config, files)
     return list(zip(files, [label]*len(files)))
 
+def _record_cache_contents(cfg, files):
+    '''Writes cache filenames to log for safe removal using 'clear_cache()' '''
+    with open(cfg.cache_dir/"cache_contents.txt", 'a+') as f:
```
Comment (mogwai) [inline audio/data.py]: Try block

**Review by mogwai — CHANGES_REQUESTED** [review]

File: audio/data.py (hunk 1 above)
Comment (rbracco) [reply audio/data.py]: Good call

File: audio/data.py [hunk 4]
```diff
@@ -68,10 +68,50 @@ class AudioConfig:
...
+            for line in pb:
+                if not os.path.exists(line): continue
+                else:
+                    try:
+                        os.remove(line)
+                    except Exception as e:
+                        print(f"Warning: Failed to remove {line}, due to error {str(e)}...continuing")
+                    parent = Path(line).parents[0]
+                    parent_dirs.add(parent)
+                    num_removed += 1
```
Comment (mogwai) [inline audio/data.py]: Move into the try block?

File: audio/data.py [hunk 5]
```diff
@@ -68,10 +68,51 @@ class AudioConfig:
...
+                if not os.path.exists(line): continue
+                else:
+                    try:
+                        os.remove(line)
+                    except Exception as e:
+                        print(f"Warning: Failed to remove {line}, due to error {str(e)}...continuing")
+                    else:
+                        parent = Path(line).parents[0]
+                        parent_dirs.add(parent)
+                        num_removed += 1
```
Comment (mogwai) [inline audio/data.py:96]: You want:
```
 try:
                        os.remove(line)
                        parent = Path(line).parents[0]
                        parent_dirs.add(parent)
                        num_removed += 1
                    except Exception as e:
                        print(f"Warning: Failed to remove {line}, due to error {str(e)}...continuing")         
```

**Review by mogwai — CHANGES_REQUESTED** [review]

File: audio/data.py (hunk 5 above)
Comment (mogwai) [reply audio/data.py:96]: elkse doesn't match with an if statement

**Pull request merged**